epan/dissectors/packet-ssh.c

```diff
@@ -580,21 +604,38 @@ dissect_ssh(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, void* data _U_)
         global_data->key1_len = len1;
         global_data->peer_data[CLIENT_PEER_DATA].mac_length=-1;
         global_data->peer_data[SERVER_PEER_DATA].mac_length=-1;
+        global_data->dest = pinfo->dst;
+        global_data->src =  pinfo->src;
```

Comment:
dst/src are volatile, if you need to keep them, you have to make a copy. See for example
https://github.com/wireshark/wireshark/blob/40d822ed61831733e5595dbb57bf4e63da856e50/epan/dissectors/packet-wireguard.c#L1041-L1054
https://github.com/wireshark/wireshark/blob/40d822ed61831733e5595dbb57bf4e63da856e50/epan/dissectors/packet-wireguard.c#L1075-L1076